src/Controllers/UserController.php

```diff
@@ -35,31 +35,46 @@ public function setService(AbstractService $userService)
      */
     public function listAll(RequestInterface $request)
     {
+        // Check if user has permission
         if ($this->session->get('role') !== 'admin') {
             return $this->getRedirectPage('/');
         }
 
         $renderParams = [
             'message' => $this->session->get('message'),
-            //TODO Extract needed params like message
             'session' => $this->session,
-            'role' => $request->getParameter('role'),
-            'email' => $request->getParameter('email'),
-            'sort' => $request->getParameter('sort')
         ];
         $this->session->delete('message');
 
+        $sortParam = ($request->getParameter('sort')) ?? "";
```

Comment:
a lot of paginator logic..

How about you add a static factory method on `Paginator` called `createFromRequest(Request $request): Paginator` or a factory class `PaginatorFactory` with `createFromRequest` and delegate creating the paginator to that